src/Lib.hs

```diff
@@ -133,6 +232,28 @@ insertPet p roster = case roster of
   _ -> error "OH NO"
 
 
+-- TODO update to account for healthRemaining
+-- such that it will get the first pet with healthRemaining > 0
+-- probably leave this as 'generic' and add a 'specialized' version
+-- just like 'keepAsking'
+getRosterFirst :: Roster -> Maybe Pet
+getRosterFirst roster = case roster of
+  Roster (Just p) _ _ _ _ _ -> Just p
+  Roster Nothing (Just p) _ _ _ _ -> Just p
+  Roster Nothing Nothing (Just p) _ _ _ -> Just p
+  Roster Nothing Nothing Nothing (Just p) _ _ -> Just p
+  Roster Nothing Nothing Nothing Nothing (Just p) _ -> Just p
+  Roster Nothing Nothing Nothing Nothing Nothing (Just p) -> Just p
+  Roster Nothing Nothing Nothing Nothing Nothing Nothing -> Nothing -- says this line is redundant?
```

Comment:
Does it still say this is redundant?

Reply:
No no more warning - forgot to delete the comment